### PR TITLE
[Master] Fix cookie path resolution logic

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6788

## Examples

N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~
- [ ] ~Checked the impact on OpenAPI generation~

### CODE REVIEW

File: ballerina-tests/http-advanced-tests/tests/http_cookies_test.bal [hunk 1]
```diff
@@ -723,3 +795,46 @@ public function testCaseSensitiveDomain() returns error? {
...
+    httpClient = check new (string `localhost:${cookieTestPort2}/cookies`, cookieConfig = {enabled: true}, secureSocket = {enable: false});
+    http:Response _ = check httpClient->get("");
+    result = check httpClient->get("");
+    test:assertEquals(result, "ballerina");
+
+    httpClient = check new (string `localhost:${cookieTestPort2}/cookies/path`, cookieConfig = {enabled: true}, secureSocket = {enable: false});
+    http:Response _ = check httpClient->get("");
+    result = check httpClient->get("");
+    test:assertEquals(result, "ballerina");
+}
```
Comment (ThisaruGuruge) [inline ballerina-tests/http-advanced-tests/tests/http_cookies_test.bal:840]:
```suggestion
}

```

**Review by ThisaruGuruge — DISMISSED** [review]
LGTM

File: ballerina/cookie_cookieStore.bal [hunk 2]
```diff
@@ -382,21 +383,37 @@ const string URL_TYPE_2 = "http://www.";
...
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_3.length(), url.length()))[0];
     } else if url.startsWith(URL_TYPE_4) {
-        domain = url.substring(URL_TYPE_4.length(), url.length());
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_4.length(), url.length()))[0];
     }
     return domain.toLowerAscii();
 }
 
+// Construct the request path from client url and the path from client method invocation.
+isolated function getReqPath(string url, string path) returns string {
```
Comment (MohamedSabthar) [inline ballerina/cookie_cookieStore.bal:404]: we could use regex here.
```
ballerina
function getReqPath(string url, string path) returns string {
    regexp:Groups? groups = re `^(?:https?://)?[^/]+(/.*)?$`.findGroups(url);
    if groups is () || groups.length() == 1 || groups[1] is () {
        return path;
    }
    return (<regexp:Span>groups[1]).substring() + path;
}
```

Comment (TharmiganK) [reply ballerina/cookie_cookieStore.bal:404]: Fixed via [0ae35c3](https://github.com/ballerina-platform/module-ballerina-http/pull/2091/commits/0ae35c346e351278c50a5c9e32c7b32393a4c1c5)

File: ballerina-tests/http-advanced-tests/tests/http_cookies_test.bal (hunk 1 above)
Comment (TharmiganK) [reply ballerina-tests/http-advanced-tests/tests/http_cookies_test.bal]: Fixed via [0ae35c3](https://github.com/ballerina-platform/module-ballerina-http/pull/2091/commits/0ae35c346e351278c50a5c9e32c7b32393a4c1c5)

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**